arbitrator/vid-helper/src/namespace.rs

```diff
@@ -0,0 +1,445 @@
+            start..end
+        };
+
+        // parse tx_offset bytes from tx table
+        let mut tx_offset_bytes = [0u8; TxTableEntry::byte_len()];
+        tx_offset_bytes[..tx_offset_range.len()].copy_from_slice(&ns_bytes[tx_offset_range]);
+        usize::try_from(TxTableEntry::from_bytes(&tx_offset_bytes).unwrap_or(TxTableEntry::zero()))
+            .unwrap_or(0)
+    }
+}
+
+impl TxTableEntry {
+    pub const MAX: TxTableEntry = Self(TxTableEntryWord::MAX);
+
+    /// Adds `rhs` to `self` in place. Returns `None` on overflow.
+    pub fn checked_add_mut(&mut self, rhs: Self) -> Option<()> {
+        self.0 = self.0.checked_add(rhs.0)?;
+        Some(())
+    }
+    pub const fn zero() -> Self {
+        Self(0)
+    }
+    pub const fn one() -> Self {
+        Self(1)
+    }
+    pub const fn to_bytes(&self) -> [u8; size_of::<TxTableEntryWord>()] {
+        self.0.to_le_bytes()
+    }
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        Some(Self(TxTableEntryWord::from_le_bytes(
+            bytes.try_into().ok()?,
+        )))
+    }
+    /// Infallible constructor.
+    pub fn from_bytes_array(bytes: [u8; TxTableEntry::byte_len()]) -> Self {
+        Self(TxTableEntryWord::from_le_bytes(bytes))
+    }
+    pub const fn byte_len() -> usize {
+        size_of::<TxTableEntryWord>()
+    }
+
+    pub fn from_usize(val: usize) -> Self {
+        Self(
+            val.try_into()
+                .expect("usize -> TxTableEntry should succeed"),
+        )
+    }
+}
+
+impl fmt::Display for TxTableEntry {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl TryFrom<usize> for TxTableEntry {
+    type Error = <TxTableEntryWord as TryFrom<usize>>::Error;
+
+    fn try_from(value: usize) -> Result<Self, Self::Error> {
+        TxTableEntryWord::try_from(value).map(Self)
+    }
+}
+impl TryFrom<TxTableEntry> for usize {
+    type Error = <usize as TryFrom<TxTableEntryWord>>::Error;
+
+    fn try_from(value: TxTableEntry) -> Result<Self, Self::Error> {
+        usize::try_from(value.0)
+    }
+}
+
+impl TryFrom<VmId> for TxTableEntry {
+    type Error = <TxTableEntryWord as TryFrom<u64>>::Error;
+
+    fn try_from(value: VmId) -> Result<Self, Self::Error> {
+        TxTableEntryWord::try_from(value.0).map(Self)
+    }
+}
+impl TryFrom<TxTableEntry> for VmId {
+    type Error = <u64 as TryFrom<TxTableEntryWord>>::Error;
+
+    fn try_from(value: TxTableEntry) -> Result<Self, Self::Error> {
+        Ok(Self(From::from(value.0)))
+    }
+}
+
+pub type VidScheme = jf_primitives::vid::advz::Advz<ark_bls12_381::Bls12_381, sha2::Sha256>;
+
+/// Namespace proof type
+///
+/// # Type complexity
+///
+/// Jellyfish's `LargeRangeProof` type has a prime field generic parameter `F`.
+/// This `F` is determined by the pairing parameter for `Advz` currently returned by `test_vid_factory()`.
+/// Jellyfish needs a more ergonomic way for downstream users to refer to this type.
```

Comment:
Is there an open issue for this? If not, create one and link it here?

Reply:
This is all copy and pasted from the sequencer repo for now, eventually we should do away with this repo entirely when we get to: https://github.com/EspressoSystems/nitro-espresso-integration/issues/87. Having all of the namespace logic here is an unfortunate workaround until the sequencer VID logic is wasm-compatible. I'll add a comment explaining this.